### ISSUE TITLE
Errors in debug mode

### ISSUE THREAD

**Issue by wjaskowski** [neptune-ai/neptune-client#555]
Some regression in the last release `0.9.9` for mode='debug' since I am now getting errors:
```
  File "/home/wojciech/miniconda3/envs/nori/lib/python3.8/site-packages/prime/config/decorators.py", line 423, in _fn
    return fn(*bound_args.args, **bound_args.kwargs)
  File "/home/wojciech/projects/nori/run/agent/evaluate.py", line 92, in evaluate
    run["params"] = flatten_dict(cfg, sep=".")
  File "/home/wojciech/miniconda3/envs/nori/lib/python3.8/site-packages/neptune/new/run.py", line 123, in __setitem__
    self.__getitem__(key).assign(value)
  File "/home/wojciech/miniconda3/envs/nori/lib/python3.8/site-packages/neptune/new/handler.py", line 104, in assign
    self[key].assign(value, wait)
  File "/home/wojciech/miniconda3/envs/nori/lib/python3.8/site-packages/neptune/new/handler.py", line 102, in assign
    return self._assign_impl(value, wait)
  File "/home/wojciech/miniconda3/envs/nori/lib/python3.8/site-packages/neptune/new/handler.py", line 112, in _assign_impl
    self._run.define(self._path, value, wait)
  File "/home/wojciech/miniconda3/envs/nori/lib/python3.8/site-packages/neptune/new/run.py", line 315, in define
    attr.assign(value, wait)
  File "/home/wojciech/miniconda3/envs/nori/lib/python3.8/site-packages/neptune/new/attributes/atoms/boolean.py", line 30, in assign
    self._enqueue_operation(AssignBool(self._path, value.value), wait)
  File "/home/wojciech/miniconda3/envs/nori/lib/python3.8/site-packages/neptune/new/attributes/attribute.py", line 39, in _enqueue_operation
    self._run._op_processor.enqueue_operation(operation, wait)
  File "/home/wojciech/miniconda3/envs/nori/lib/python3.8/site-packages/neptune/new/internal/operation_processors/sync_operation_processor.py", line 34, in enqueue_operation
    raise errors[0]
  File "/home/wojciech/miniconda3/envs/nori/lib/python3.8/site-packages/neptune/new/internal/backends/neptune_backend_mock.py", line 136, in execute_operations
    self._execute_operation(run_uuid, op)
  File "/home/wojciech/miniconda3/envs/nori/lib/python3.8/site-packages/neptune/new/internal/backends/neptune_backend_mock.py", line 156, in _execute_operation
    run.pop(op.path)
  File "/home/wojciech/miniconda3/envs/nori/lib/python3.8/site-packages/neptune/new/internal/run_structure.py", line 66, in pop
    self._pop_impl(self._structure, path, path)
  File "/home/wojciech/miniconda3/envs/nori/lib/python3.8/site-packages/neptune/new/internal/run_structure.py", line 82, in _pop_impl
    self._pop_impl(ref[head], tail, attr_path)
  File "/home/wojciech/miniconda3/envs/nori/lib/python3.8/site-packages/neptune/new/internal/run_structure.py", line 74, in _pop_impl
    raise MetadataInconsistency("Cannot delete {}. Attribute not found.".format(path_to_str(attr_path)))
neptune.new.exceptions.MetadataInconsistency: Cannot delete params/agent.pair_q_single_max_exp.make_symmetric. Attribute not found.
Result for evaluate_bf4ff_00002:
  {}
```

This was working before. Everything works fine with a normal backend.

**Comment by aniezurawski** [comment]
Hi @wjaskowski 

What version have you been using before?
Could you provide reproduction steps?

**Comment by aniezurawski** [comment]
I tried with such minimal code snippet and failed to reproduce the issue.
```
import neptune.new as neptune

run = neptune.init(mode="debug")

run["x"] = 5
del run["x"]
```

**Comment by wjaskowski** [comment]
```
import neptune.new as neptune
run = neptune.init(mode="debug")
run["params"] = { 'predictor.cheat': False}
```
Bug introduced in 0.9.6

**Comment by aniezurawski** [comment]
Fixed in 0.9.10.
